content/docs/user-guide/programming/components/reflection/behavior-context.md

```diff
@@ -0,0 +1,521 @@
+---
+linkTitle: Behavior Context
+title: Behavior Context in O3DE
+description: Use the behavior context in Open 3D Engine (O3DE) to make runtime code accessible to scripting systems such as Script Canvas and Lua.
+weight: 300
+---
+
+The behavior context makes runtime code accessible to **Open 3D Engine (O3DE)** scripting systems such as Script Canvas or Lua. It provides script bindings that invoke runtime C++ methods, read constants, write properties, and create and handle **Event Bus (Ebus)** events. You can have multiple behavior contexts that are specialized for different purposes, and you can unreflect the behavior contexts to implement reloading.
+
+Use the behavior context to bind the following C++ constructs for scripting:
+
++ [Classes](#classes)
++ [Methods](#methods)
++ [Properties](#properties)
++ [Constants](#constants)
++ [Enums](#enums)
+
+In addition, the behavior context supports the O3DE [EBus](/docs/user-guide/programming/ebus) and [AZ::Event](/docs/user-guide/programming/az-event) event systems:
+
++ [EBus](#ebus)
+  + [Events](#events)
+  + [Event Handlers](#event-handlers)
+
+## Classes
+
+Classes in the behavior context reflect a C++ class or struct. You can provide an optional name for a class. If you do not provide a name, the class name from `AzTypeInfo` is used. That name must be unique for the scope. Because the system uses `AzRTTI` to build the class hierarchy, you can use RTTI if you want to reflect base class functionality.
```

Comment:
```suggestion
Classes in the behavior context reflect a C++ class or struct. You can provide an optional name for a class. If you don't provide a name, the class name from `AzTypeInfo` is used. That name must be unique for the scope. Because the system uses `AzRTTI` to build the class hierarchy, you can use RTTI if you want to reflect base class functionality.
```
Change for consistency with other occurrences of "don't" in the topic.